src/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```

Comment:
- [ ] the `@nameable` should return it's corrected name. so calling `@nameable.correct_name` is necessary.